articles/140_topic_and_service_name_mapping.md

```diff
@@ -219,13 +201,94 @@ Since ROS 2 topic and service names are expanded to fully qualified names, any b
 Additionally any URL related syntax, e.g. the `rostopic://` prefix, will be removed once parsed.
 Therefore only forward slashes have to be substituted when converting to DDS topic names.
 
-### Substitution of the Namespace Delimiter
+### DDS Partitions as namespace hierarchy
 
-The namespace delimiter in ROS 2 topic and service names, a forward slash (`/`), will be replaced with double underscores (`__`).
-Note that as fully qualified ROS 2 topic and service names are absolute, there is always a leading forward slash (`/`).
+We chose to implement the ROS namespace hierarchy with the help of DDS Partitions.
+DDS Paritions are part of the Subscription/Publisher QoS settings and have been considered as the prefered method of implementing namespaces.
+A subscriber and publisher only match when not only their basename topic matches, they also have to be specified on the same partiton.
+The big benefit of handling the namespaces with partitions is that it is an official QoS within the DDS specification.
+Furthermore, we can separate a ROS topic hierarchy from a DDS implementation, by forwarding the ROS topic in form of `/ns1/ns2/ns3/base_name` to the DDS implementation, which then takes take about setting up a hierarchy.
+All this stays generally ROS agnostic and open to alternative implementations of the RMW interface (e.g. ZeroMQ etc.), which may want to introduce another, different namespace system.
+
+#### DDS Partitions Characteristics
+
+A DDS topic at this point is the basename token within a ROS topic.
+The namespace of a ROS topic is implemented within the partition field of DDS.
+DDS partitions are implemented as a string array within the Subscription/Publisher QoS, allowing up to 64 items.
+The maximum size of each string has yet to be verified and may depend strongly on the implementation.
+Hereby, the partitions field is implemented as an array of strings, where each string is an individual partitions.
+This means further that every partition index is stricly unique and independent from all other items.
+Special character used within regular expressions (such as `+`, `*`, `^`) are officially not supported by the standard.
+
+DDS Partitions don't support a partition hierarchy.
+Each entry in the partition array is directly combined with the DDS topic and they are not sequentially combined.
+If a publisher operates on two partition entries, e.g. `foo` and `bar` with a base name of `baz`, the resulting topic will be twofold:
+
+```
+- /foo/baz
+- /bar/baz
+```
+
+That implies, that DDS Partitions by default are not depicting any hierarchy.
+On the other hand, it allows an easy aliasing or duplication of the specified topic, given that the basename topic does not change.
+A dynamic duplication of the original topic such as `cmd_vel` in a namespace `robot1` to `robot2/cmd_vel` can thus be easily achieved by expanding the DDS partition array.
```

Comment:
This aliasing approach will only work if only the namespace is different. If the topic name needs to be different this will likely require multiple separate DDS publishers. Therefore I am not sure that we should describe this here. This is an implementation detail and I am not convinced that we should commit to this implementation option (since we will likely have to implement another option anyway).